mat64/vector.go

```diff
@@ -71,11 +71,11 @@ func NewVector(n int, data []float64) *Vector {
 }
 
 // ViewVec returns a sub-vector view of the receiver starting at element i and
-// extending n columns. If i is out of range, or if n is zero or extend beyond the
-// bounds of the Vector ViewVec will panic with ErrIndexOutOfRange. The returned
-// Vector retains reference to the underlying vector.
+// extending n rows. If i is out of range, or if n is zero, or if the view
+// extends beyond the bounds of the Vector, ViewVec will panic with ErrIndexOutOfRange.
+// The returned Vector retains reference to the underlying vector.
 func (v *Vector) ViewVec(i, n int) *Vector {
-	if i+n > v.n {
+	if i < 0 || n <= 0 || i+n > v.n {
 		panic(matrix.ErrIndexOutOfRange)
 	}
 	return &Vector{
```

Comment:
Can you add an upper bound to the Data slice. When we add the overlap detection to Vector operations it will detect things as overlapping if they are not properly truncated. (We'll probably detect other cases of this when I wire that in).